core-models/src/main/scala/com/mesosphere/usi/core/models/PodSpec.scala

```diff
@@ -13,3 +15,45 @@ package com.mesosphere.usi.core.models
   * @param runSpec WIP the thing to run, and resource requirements, etc.
   */
 case class PodSpec(id: PodId, goal: Goal, runSpec: RunSpec)
+
+object PodSpec {
+  type ValidationMessage = String
+
+  /**
+    * Verifies that every value in range is requested only once. Requesting same value multiple times would yield pod that will never be scheduled.
+    * E.g. requesting two ports 80 on one machine is just not possible to satisfy
+    * @param runSpec runSpec we are validating
+    * @return true if provided range requirements are valid
+    */
+  private def validateStaticRangeRequirementsUnique(runSpec: RunSpec): Seq[ValidationMessage] = {
+    val staticPorts = runSpec.resourceRequirements.collect {
+      case RangeRequirement(requestedValues, _, _) => requestedValues
+    }.flatten.collect { case ExactValue(value) => value }
+
+    if (staticPorts.distinct.length != staticPorts.length) {
+      Seq(s"Every value inside RangeResource can be requested only once. Requirement: ${staticPorts.mkString(",")}")
+    } else {
+      Seq.empty
```

Comment:
What do you think about `val Valid = Seq.empty`? So that we have 
```suggestion
      Valid
```

Reply:
push

Reply:
👍